functions/New-DbaDatabase.ps1

```diff
@@ -0,0 +1,317 @@
+                #add the log to the db
+                $NewDB.LogFiles.Add($TLog)
+            }
+            catch
+            {
+                Stop-Function -Message "Error adding log file to database." -ErrorRecord $_ -Target $instance -Continue
+            }
+
+            #set database settings; collation, owner, recovery model
+
+            #set the collation
+            if ($Collation.Length -eq 0)
+            {
+                $Message = "USing default server collation"
+                Write-Verbose $Message
+            }
+            else
+            {
+                $Message = "Setting collation to $Collation"
+                Write-Verbose $Message
+                $NewDB.Collation = $Collation
+            }
+
+            #set the recovery model
+            if ($RecoveryModel.Length -eq 0)
+            {
+                $Message = "Using default recovery model from the Model database"
+                Write-Verbose $Message
+            }
+            else
+            {
+                Write-Message -message "Setting recovery model to $RecoveryModel" -level Verbose
+                $NewDB.RecoveryModel = $RecoveryModel
+            }
+
+            #we should now be able to create the db, and run any other config settings afterwards
+            Write-Message -message "Creating Datbase $DatabaseName" -level verbose
+            if ($PSCmdlet.ShouldProcess($instance, "Creating the database on $instance")) 
+            {
+                try
+                {
+                    $NewDb.Create()
+                }
+                catch
+                {
+                    Stop-Function -Message "Error creating Database $DatabaseName on server $SqlInstance" -ErrorRecord $_ -Target $instance -Continue
+                }      
+
+
+                #now do post db creation work, set the dbowner and set the default filegroup
+                #Set the owner
+                Write-Message -message "Setting database owner to $DatabaseOwner" -level verbose
+                try
+                {
+                    $NewDB.SetOwner($DatabaseOwner)
+                }
+                catch
+                {
+                    Stop-Function -Message "Error setting Database Owner to $DatabaseOwner" -ErrorRecord $_ -Target $instance -Continue
+                }
+
+                #set the user filegroup to be the default
+                write-message -message "Setting default filegroup to $UserFileGroupName" -level verbose
+                try
+                {
+                    $NewDB.SetDefaultFileGroup($UserFileGroupName)
+                }
+                catch
+                {
+                    Stop-Function -Message "Error setting default filegorup to $UserFileGroupName" -ErrorRecord $_ -Target $instance -Continue
+                }
+
+            }
+
+            #Write completed message
+            Write-Message -message "Completed creating database $DatabaseName on server $sqlInstance" -level Verbose
```

Comment:
Use `$instance` variable instead for the message content. `$sqlinstance` may be more than one instance